context-ruby/spec/trunk_spec_helper.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
```

Comment:
There are 2 primary components here. One is a custom formatter which is able to parse out the final results. The other is an rspec helper that monitors the test state and modifies the metadata to track values of interest (attempts, quarantining, etc...).

RSpec helpers do not have access to final results because they are considered apart of the run. The way this is defined automatically links the formatter to the test run.